src/mg/PAMI/Client/Impl/ClientImpl.php

```diff
@@ -405,20 +405,21 @@ public function send(OutgoingMessage $message)
 	    if (@fwrite($this->_socket, $messageToSend) < $length) {
     	    throw new ClientException('Could not send message');
 	    }
-	    $read = 0;
-	    while($read <= $this->_rTimeout) {
-	        $this->process();
```

Comment:
Ok understood.
So that means that rTimeout is the amount of time to wait for writing and/or reading from the socket. 0 meaning indefinitely ? If so the while loop should be rethought, once we start receiving something we need to wait for the EOM and handle any error if/when it occurs.)